### PR TITLE
#522 course configs from url

### PR DESCRIPTION
**Pull request by jaakkonarhi**
# Description of the PR

+ [https://version.aalto.fi/gitlab/aplus-courses/course-config-urls/-/blob/main/courses.yaml ](https://version.aalto.fi/gitlab/aplus-courses/course-config-urls/-/blob/main/courses.yaml)

# Testing
<table>
  <tr>
    <th>unit</th>
    <th>integration</th>
    <th>manual</th>
  </tr>
  <tr>
    <td>
      <ul>
        <li>- [ ] new <b>unit</b> tests created</li>
        <li>- [ ] all <b>unit</b> tests pass</li>
      </ul>
    </td>
    <td>
      <ul>
        <li>- [ ] new <b>integration</b> tests created</li>
        <li>- [ ] all <b>integration</b> tests pass</li>
      </ul>
    </td>
    <td>
      <ul>
        <li>- [ ] <b>manual</b> testing went well</li>
      </ul>
    </td>
  </tr>
</table>  
  
# Have you updated the [TESTING.md](https://github.com/Aalto-LeTech/aplus-courses/blob/master/TESTING.md) or other relevant documentation on _your_ branch?

- [ ] Yes.
- [ ] Not yet. I will do it next.
- [x] Not relevant.

### CODE REVIEW

File: src/main/java/fi/aalto/cs/apluscourses/intellij/actions/CourseProjectAction.java [hunk 1]
```diff
@@ -280,6 +289,25 @@ private URL tryGetCourseUrl(@NotNull Project project) {
     }
   }
 
+  private CourseItemViewModel[] fetchCourses(@NotNull Project project) {
+    try {
+      var url = new URL(COURSE_LIST_URL);
+      var courseStream = CoursesClient.fetch(url);
+      var yaml = new Yaml();
+
+      @SuppressWarnings("unchecked") var courseList = (List<Map<String, String>>) yaml.load(courseStream);
```
Comment (Taikelenn) [inline src/main/java/fi/aalto/cs/apluscourses/intellij/actions/CourseProjectAction.java:298]: This looks "non-object-oriented" but from what I've found it's the simplest way. Other methods that look "cleaner" use some ClassLoaders or a metric ton of boilerplate code.

**Review by Taikelenn — APPROVED** [review]
obraz
Could the text be changed to something else than "Nothing to show", for example "Loading, please wait..."? Even though most users won't even notice it, the one or two users who are still using dial-up in 2022 and experience latency measured in minutes would certainly appreciate it.

Other than that, it works fine.

**Review by stellatsv — APPROVED** [review]

**Review by OlliKiljunen — COMMENTED** [review]
I see you got two approvals so my review isn't needed. :)

**Comment by jaakkonarhi** [comment]
I put a loading icon on the course list, good enough. There should be no problems even if we add new fields to the yaml

**Pull request merged**